PostService/PostService.Repository/PostRepository.cs

```diff
@@ -33,23 +33,69 @@ public async Task<PagedList<Post>> FindAll(PaginationParams paginationParams)
                                                paginationParams.PageSize);
         }
 
-        public async Task<PagedList<Post>> FindAllPublicAndFollowed(PaginationParams paginationParams)
+        
+
+        public async Task<PagedList<Post>> FindAllPublic(PaginationParams paginationParams)
         {
-            var connections = _context.Connections;
             var publicPosts = from post in _context.Posts
-                      join profile in _context.Profiles on post.AuthorId equals profile.Id
-                      where profile.Public == true
+                              join profile in _context.Profiles on post.AuthorId equals profile.Id
+                              where profile.Public == true
+                              select post;
+
+            return PagedList<Post>.ToPagedList(publicPosts,
+                                               paginationParams.PageNumber,
+                                               paginationParams.PageSize);
+        }
+
+        public async Task<PagedList<Post>> FindAllFollowed(PaginationParams paginationParams, Guid profileId)
+        {
+            var connections = from c in _context.Connections
+                              select c;
+
+            var profile1 = from post in _context.Posts
+                           join connection in connections on post.AuthorId equals connection.Profile1
+                           where post.AuthorId == profileId
+                           select connection.Profile2;
+
+            var profile2 = from post in _context.Posts
+                           join connection in connections on post.AuthorId equals connection.Profile2
+                           where post.AuthorId == profileId
+                           select connection.Profile1;
+
+            var connectedProfiles = profile1.Concat(profile2);
+            var res = from profile in connectedProfiles
+                      join post in _context.Posts on profile equals post.AuthorId
                       select post;
-                  
-            var profile1 = from post in publicPosts
+
+            return PagedList<Post>.ToPagedList(res,
+                                               paginationParams.PageNumber,
+                                               paginationParams.PageSize);
+        }
+
+        public async Task<PagedList<Post>> FindAllPublicAndFollowed(PaginationParams paginationParams, Guid profileId)
```

Comment:
Jednom metodom dobavljamo i postove javnih i pracenih profila, jel tako?
Ako jeste, mozda bi bilo lakse samo da pozoves ove dve metode koje si vec napisao i spojis rezultat.

Reply:
Bilo bi lakse, ali ne mogu da ih spajam jer je njihova povratna vrednost _PagedList_, a to je custom klasa.